src/coreclr/jit/importer.cpp

```diff
@@ -2872,48 +2879,35 @@ GenTree* Compiler::impCloneExpr(GenTree*             tree,
     // specialized type (e.g. a SIMD type).  So we will get the type from
     // the lclVar AFTER calling impAssignTempGen().
 
-    impAssignTempGen(temp, tree, structHnd, curLevel, pAfterStmt, impCurStmtOffs);
+    impAssignTempGen(temp, tree, structHnd, curLevel, pAfterStmt, impCurStmtDI);
     var_types type = genActualType(lvaTable[temp].TypeGet());
 
     *pClone = gtNewLclvNode(temp, type);
     return gtNewLclvNode(temp, type);
 }
 
-/*****************************************************************************
- * Remember the IL offset (including stack-empty info) for the trees we will
- * generate now.
- */
-
-inline void Compiler::impCurStmtOffsSet(IL_OFFSET offs)
+DebugInfo Compiler::impCreateDIWithCurrentStackInfo(IL_OFFSET offs, bool isCall)
 {
-    if (compIsForInlining())
-    {
-        Statement* callStmt = impInlineInfo->iciStmt;
-        impCurStmtOffs      = callStmt->GetILOffsetX();
-    }
-    else
-    {
-        assert(offs == BAD_IL_OFFSET || (offs & IL_OFFSETX_BITS) == 0);
-        IL_OFFSETX stkBit = (verCurrentState.esStackDepth > 0) ? IL_OFFSETX_STKBIT : 0;
-        impCurStmtOffs    = offs | stkBit;
-    }
+    assert(offs != BAD_IL_OFFSET);
+
+    bool isStackEmpty = verCurrentState.esStackDepth <= 0;
+    return DebugInfo(compInlineContext, ILLocation(offs, isStackEmpty, isCall));
 }
 
 /*****************************************************************************
- * Returns current IL offset with stack-empty and call-instruction info incorporated
+ * Remember the IL offset (including stack-empty info) for the trees we will
```

Comment:
Update this comment block to new-style?

Reply:
Will do.

Related to this function, I have been thinking that as a future change we might want to unconditionally record DebugInfo for all statements created during import and instead record all necessary information to determine much later (e.g. in the emitter) whether we should actually communicate that particular debug info back to the runtime. For instance, we would record all the different kinds of boundaries that the runtime can ask for as information in the `ILLocation` and then query the requested boundaries in the emitter instead.

The benefit would be that we can track debug information diffs in a much more "stable" way that does not depend on what the runtime tells us at a particular time. One challenge is with `GenTreeCall` nodes where we use a map on the side that is only populated when the runtime actually asks for boundaries to support the "method X returned Y" feature.